src/main.rs

```diff
@@ -17,22 +17,44 @@ use vulkano::swapchain::PresentMode;
 use settings::WasabiSettings;
 use state::WasabiState;
 use winit::{
+    dpi::{LogicalSize, Size},
     event::{Event, WindowEvent},
     event_loop::{ControlFlow, EventLoop},
     window::Fullscreen,
 };
 
+pub const WINDOW_SIZE: Size = Size::Logical(LogicalSize {
+    width: 1280.0,
+    height: 720.0,
+});
+
 pub fn main() {
     // Winit event loop
     let event_loop = EventLoop::new();
 
     // Load the settings values
     let mut settings = WasabiSettings::new_or_load();
     let mut wasabi_state = WasabiState::default();
+    #[cfg(unix)]
+    let is_wayland = winit::platform::unix::EventLoopWindowTargetExtUnix::is_wayland(
+        std::ops::Deref::deref(&event_loop),
+    );
 
     // Create renderer for our scene & ui
-    let window_size = [1280, 720];
-    let mut renderer = Renderer::new(&event_loop, window_size, PresentMode::Immediate, "Wasabi");
+    let mut renderer = Renderer::new(
+        &event_loop,
+        #[cfg(unix)]
+        if is_wayland {
+            println!("Present Mode: Mailbox");
+            PresentMode::Mailbox
+        } else {
+            println!("Present Mode: Immediate");
+            PresentMode::Immediate
+        },
+        #[cfg(not(unix))]
+        PresentMode::Immediate,
+        "Wasabi",
+    );
```

Comment:
I'd personally prefer if all of this `#[cfg(unix)]` stuff would be extracted into a separate function which returns something like the present mode, or another enum which describes which present modes should be chosen from.